apiserver/facades/client/bundle/bundle.go

```diff
@@ -152,9 +157,142 @@ func (b *BundleAPI) GetChanges(args params.BundleChangesParams) (params.BundleCh
 
 // ExportBundle exports the current model configuration as bundle.
 func (b *BundleAPI) ExportBundle() (params.StringResult, error) {
-	return params.StringResult{}, nil
+	var fail = func(failErr error) (params.StringResult, error) {
+		return params.StringResult{}, common.ServerError(failErr)
+	}
+
+	if err := b.checkCanRead(); err != nil {
+		return fail(err)
+	}
+
+	exportConfig := b.backend.GetExportConfig()
+	model, err := b.backend.ExportPartial(exportConfig)
+	if err != nil {
+		return fail(err)
+	}
+
+	// Fill it in charm.BundleData datastructure.
+	bundleData, err := b.fillBundleData(model)
+	if err != nil {
+		return fail(err)
+	}
+
+	bytes, err := yaml.Marshal(bundleData)
+	if err != nil {
+		return fail(err)
+	}
+
+	return params.StringResult{
+		Result: string(bytes),
+	}, nil
 }
 
 // Mask the new method from V1 API.
 // ExportBundle is not in V1 API.
 func (u *APIv1) ExportBundle() (_, _ struct{}) { return }
+
+// FillBundle fills the bundledata datastructure required for the exportBundle.
+func (b *BundleAPI) fillBundleData(model description.Model) (*charm.BundleData, error) {
+	// get the default-series from model config.
+	cfg := model.Config()
+	value, ok := cfg["default-series"]
+	if !ok {
+		value = series.LatestLts()
+	}
+	defaultSeries := fmt.Sprintf("%v", value)
+
+	data := &charm.BundleData{
+		Series:       defaultSeries,
+		Applications: make(map[string]*charm.ApplicationSpec),
+		Relations:    [][]string{},
+		Machines:     make(map[string]*charm.MachineSpec),
+	}
+
+	if len(model.Applications()) == 0 {
+		return &charm.BundleData{}, errors.Errorf("nothing to export as there are no applications.")
```

Comment:
Error messages should not contain a full stop as other layers may append more information to the message.

It's kind of counter-intuitive until you see the trace coming from cmd layer :)